tutorial/Flyer_Basics.ipynb

```diff
@@ -0,0 +1,985 @@
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "('ab9f6cd6-2ed2-420c-8dbb-d65adedb0add',)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "class BareMinimumFlyer(ophyd.Device):\n",
+    "\n",
+    "    def kickoff(self):\n",
+    "        kickoff_status = ophyd.DeviceStatus(self)\n",
+    "        kickoff_status._finished(success=True)\n",
+    "        return kickoff_status\n",
+    "\n",
+    "    def complete(self):\n",
+    "        complete_status = ophyd.DeviceStatus(self)\n",
+    "        complete_status._finished(success=True)\n",
+    "        return complete_status\n",
+    "\n",
+    "    def collect(self):\n",
+    "        yield {'data':{}, 'timestamps':{}, 'time':time.time()}\n",
+    "    \n",
+    "    def describe_collect(self):\n",
+    "        return {self.name: {}}\n",
+    "\n",
+    "\n",
+    "flyer = BareMinimumFlyer(name=\"flyer\")\n",
+    "print(flyer.complete())\n",
+    "print(list(flyer.collect()))\n",
+    "\n",
+    "# if this next step succeeds, it's proof that we did this right!\n",
+    "RE(bluesky.plans.fly([flyer]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Flyer : a starting template <a class=\"anchor\" id=\"flyer-template\" />\n",
+    "\n",
+    "The `BareMinimumFlyer` is a good start to use a Flyer but we'll need to add a few more things to make a good template.  The first thing to do is to make the status object known to any method of the class.  We'll call it `self._completion_status` and it will tell us if the *controller* is finished.  In the constructor (`__init__()`), we set it to `None`, the value we expect when not *flying*.  Since we **need** a constructor, we must remember to call the constructor of the superclass as well or our `ophyd.Device` will not work correctly.\n",
+    "\n",
+    "    def __init__(self, *args, **kwargs):\n",
+    "        super().__init__('', parent=None, **kwargs)\n",
+    "        self._completion_status = None\n",
+    "\n",
+    "Our *controller* could take some time to signal that it is finished (seconds to minutes, at least).  We need a way to detect this completion.  We can do that either by polling the PV or by subscribing to a callback on the completion event.  We'll make that choice when we implement the actual activity.  Here, intend to wait in a polling loop.  Since the polling loop is an activity that does not return until the *controller* is done, we must do that waiting in a thread separate from that of the RunEngine.  (We do not want to block the RunEngine thread so it is free to respond to other activities, such as data from other streams or the user inerface.)  So, we run `my_activity()` in a separate thread that is called from `kickoff()`:\n",
+    "\n",
+    "        thread = threading.Thread(target=self.my_activity, daemon=True)\n",
```

Comment:
For some reasons that I actually don't understand yet, when using pyepics in a device, you need to use a `epics.ca.CAThread`. @danielballan or @tacaswell  may be able to elaborate (and I'm interested to hear).
When constructing your flyer, make sure you're using this kind of thread or else you might run into issues.

For the tutorial, we may want to add a disclaimer somewhere about its use, and warn that when calling `get()` `put()` or `set()` on an `EpicsSignal`, that we should launch the callbacks in this special type of thread.

Reply:
The issue in https://github.com/prjemian/ipython_mintvm/issues/3#issuecomment-385058577 was the beginning of this PR.  That comes from a project to develop a mock-up for this issue from our USAXS instrument (https://github.com/APS-USAXS/ipython-usaxs/issues/3).

Reply:
Part of the inspiration of this tutorial was that the existing tutorial both skips steps and combines steps.  It's more of a demo for a specific use but leaves the newcomer without an understanding of how to start on a project which differs than what is shown (such as for the [USAXS fly scan](https://github.com/prjemian/ipython_mintvm/issues/3#issuecomment-385058577)).

Reply:
Isn't the question of which threading package to be used one to be resolved in `ophyd`?  If we are to consider the impact of using an EPICS Channel Access thread manager, that should come through the `ophyd` package which is an abstraction of the underlying control system.  Otherwise, we are sure to break that code once we start to provide for other protocols such as PV Access (from EPICS v7).  I'll leave the suggestion in the tutorial for now as an alternative but continue to use `threading`.

In practice, with real EPICS PVs, no, nothing strange like that has cropped up when using the `threading` package.